dev-tools/packaging/templates/docker/Dockerfile.elastic-agent.tmpl

```diff
@@ -6,14 +6,25 @@
 # the final image because of permission changes.
 FROM {{ .buildFrom }} AS home
 
-COPY beat {{ $beatHome }}
+{{- if ne .user "root" }}
+RUN groupadd --gid 1000 {{ .BeatName }} && \
+    useradd -M --uid 1000 --gid 1000 --groups 0 {{ .user }} && \
+    true
+{{- end }}
 
-RUN mkdir -p {{ $beatHome }}/data {{ $beatHome }}/data/elastic-agent-{{ commit_short }}/logs && \
-    chown -R root:root {{ $beatHome }} && \
+COPY --chown={{ .user }}:{{ .user }} beat {{ $beatHome }}
+
+RUN true && \
+{{- if ne .user "root" }}
+    usermod -d {{ $beatHome}} {{ .user }} && \
+{{- end}}
+    # ECE needs to create config here under non-1000 user
+    chmod 0777 {{ $beatHome}} && \
+    mkdir -p {{ $beatHome }}/data {{ $beatHome }}/data/elastic-agent-{{ commit_short }}/logs && \
     find {{ $beatHome }} -type d -exec chmod 0755 {} \; && \
     find {{ $beatHome }} -type f -exec chmod 0644 {} \; && \
-    find {{ $beatHome }}/data -type d -exec chmod 0770 {} \; && \
-    find {{ $beatHome }}/data -type f -exec chmod 0660 {} \; && \
+    find {{ $beatHome }}/data -type d -exec chmod 0777 {} \; && \
+    find {{ $beatHome }}/data -type f -exec chmod 0666 {} \; && \
```

Comment:
Why do we need to create the user and the correct ownership in the intermediate stage ? Wouldn't it be enough to copy with the correct ownership and user in the last stage (line ~127) ?

Reply:
hmm that;s a nice idea, will check